source/adapters/level_zero/queue.cpp

```diff
@@ -1207,6 +1207,12 @@ ur_queue_handle_t_::executeCommandList(ur_command_list_ptr_t CommandList,
         // Turn into a more informative end-user error.
         return UR_RESULT_ERROR_UNKNOWN;
```

Comment:
@nrspruit : Just to confirm, you intended to not do this reset/event cleanup only when error is ERROR_UNKNOWN?

Reply:
Correct, given ERROR_UNKNOWN, the system is in a very bad state and even the level zero driver does not understand what is failing. with Error Unknown it is not clear if the calls to even reset etc would function at all so it is best to avoid calling for L0 apis if the error is not known.

Reply:
Got it, thanks for the explanation.